### PR TITLE
Add WikimateException and centralize API error checks in request() 

### PR DESCRIPTION
**Pull request by Xymph**
This implements the exception handling portion of #113, where the reasoning for this approach is discussed. Debug logging changes will go into a separate PR, which will close that issue.

A commit with documentation/changelog edits follows after #135 is wrapped up and merged. (Hmm, I guess this should have been a draft PR. Can it still be changed?)

### CODE REVIEW

**Comment by waldyrious** [comment]
> Hmm, I guess this should have been a draft PR. Can it still be changed?

Yep, it can 🙂. You should hopefully see a "Still in progress? Convert to draft" in the PR sidebar (right below the "Reviewers" heading).

**Comment by Xymph** [comment]
One observation about 4169410: `request()` performs a `json_decode` but returns the response object, and all callers perform another `json_decode`, which is a little inefficient. This will be addressed in the next PR that centralizes debug logging too.

File: Wikimate.php [hunk 1]
```diff
@@ -183,7 +183,8 @@ protected function initRequests()
 	 * @param  array         $headers  Optional extra headers to send with the request
 	 * @param  boolean       $post     True to send a POST request, otherwise GET
 	 * @return Requests_Response       The API response
-	 * @throw  Exception               If lagged and ran out of retries
+	 * @throw  WikimateException       If lagged and ran out of retries,
+	 *                                 or an unexpected API response
```
Comment (waldyrious) [inline Wikimate.php:187]: The second clause should include its own verb. For example:

```suggestion
	 *                                 or got an unexpected API response
```

**Review by waldyrious — APPROVED** [review]
LGTM! Just a minor grammar nit :smile:

File: Wikimate.php (hunk 1 above)
Comment (Xymph) [reply Wikimate.php]: Picky picky :smiley_cat:

**Pull request merged**